### DISCUSSION TITLE
NavigationLink binding lead to view onAppear

### DISCUSSION THREAD

**Discussion by AaronSwift-SF** [pointfreeco/swift-composable-architecture#983] [Q&A]
I have a strange problem, I need use NavigationLink push other view. If I use this method when pushed can lead to current view onAppear.
_NavigationLink(
                    isActive: viewStore.binding(get: \.isNavigationActive, send: JenkinsJobDetailAction.setNavigation(isActive:)),
                    destination: {
                        Text("...")
                    },
                    label: {
                        Image(systemName: "hammer")
                            .font(Font.system(size: 15))
                            .foregroundColor(.primary)
                    }
                )_

If I use another method, not lead to current view trigger onAppear.
_NavigationLink {
                    JenkinsJobBuildView(store: store.scope(
                        state: \.buildJob,
                        action: JenkinsJobDetailAction.buildJob
                    ))
                } label: {
                    Image(systemName: "hammer")
                        .font(Font.system(size: 15))
                        .foregroundColor(.primary)
                }_
Please help me analyze it. Thanks!

**Comment by mbrandonw** [comment]
Hi @15620946376, I'm going to convert this to a discussion since it's not an issue with the library.

**Reply by AaronSwift-SF** [reply]
ok. After testing, I found may be  NavigationLink(isActive:viewStore.binding) have some problems. If I declare @State var isNavigationActive, use it NavigationLink(isActive: $isNavigationActive), this also not lead to when pushed the current view trigger onAppear.

problem action:
received action:
  RootAction.jenkins(
    JenkinsMainAction.jobs(
      JenkinsJobsAction.detailAction(
        JenkinsJobDetailAction.setNavigation(isActive: true)
      )
    )
  )

received action:
  RootAction.jenkins(
    JenkinsMainAction.jobs(
      JenkinsJobsAction.detailAction(JenkinsJobDetailAction.onAppear)
    )
  )
